modules/nf-core/sourmash/compare/main.nf

```diff
@@ -0,0 +1,43 @@
+process SOURMASH_COMPARE {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::sourmash=4.5.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/sourmash:4.5.0--hdfd78af_0':
+        'quay.io/biocontainers/sourmash:4.5.0--hdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(signatures)
+    val save_numpy_matrix
+    val save_csv
+
+    output:
+    tuple val(meta), path("*.comp")            , optional:true, emit: numpy_matrix
+    tuple val(meta), path("*.comp.labels.txt") , optional:true, emit: labels
```

Comment:
Is the `.comp.labels.txt` really optional, shouldn't there always be output?

Reply:
Ya this is a tricky one too -- `.comp.labels.txt` is only output when `.comp` is output. The default for `sourmash compare` is to output the compare matrix to stdout. When `--output` is used, it outputs the numpy binary matrix and the labels. When `--csv` is used, it only outputs the csv file.

Reply:
Are the labels required to make sense of the matrix? In that case, I'd say the output should be a triple maybe?

```nextflow
tuple val(meta), path("*.comp.labels.txt"), path("*.comp"), optional:true, emit: matrix
```

Reply:
oh this is great, I didn't know you could do this! This is a much better suggestion.